src/FormControl/_FormControlContext.tsx

```diff
@@ -0,0 +1,59 @@
+import {createContext, useContext} from 'react'
+import {FormValidationStatus} from '../utils/types/FormValidationStatus'
+import {FormControlProps} from './FormControl'
+
+export interface FormControlContext extends Pick<FormControlProps, 'disabled' | 'id' | 'required'> {
+  captionId?: string
+  validationMessageId?: string
+  validationStatus?: FormValidationStatus
+}
+
+export const FormControlContext = createContext<FormControlContext | null>(null)
+
+interface FormControlForwardedProps extends Omit<FormControlContext, 'captionId' | 'validationMessageId'> {
+  ['aria-describedby']?: string
+}
+
+/**
+ * Make any component compatible with `FormControl`'s automatic wiring up of accessibility attributes & validation by
+ * reading the props from this hook and handling them / assigning them to the underlying form control. If used outside
+ * of `FormControl`, this hook has no effect.
+ *
+ * @param externalProps The external props passed to this component. If provided, these props will be merged with the
+ * `FormControl` props, with external props taking priority. This is also used for validating the external props,
+ * logging warnings to the console if there are conflicts.
+ */
+export function useFormControlForwardedProps<P extends FormControlForwardedProps>(externalProps: P): P {
```

Comment:
Do we like this naming? "Forwarded props" vs "external props"? It reminds me of "forwarded refs" which also come from the parent.

Reply:
I like this naming. It's clear and feels familiar.